prog/vm/nexus.rb

```diff
@@ -16,14 +16,17 @@ def self.assemble(public_key, tag_space_id, name: nil, size: "m5a.2x",
       fail "Not existing tag space"
     end
 
+    id = SecureRandom.uuid
+    name ||= uuid_to_name(id)
+
+    Validation.validate_name(name)
```

Comment:
I have doubts about where to validate name: routes(controller) or prog's assemble. I put prog assemble because it also validates name for Vm that created by developers via pry etc.

Reply:
seems reasonable. Having encountered this problem several times, I think it makes sense to raise an error that may optionally have HTTP-nature and perhaps a suggested error code when applicable (in this case, 400, is there a more specific one?)

Reply:
`Clover.error_handler` catches and redirects back with an error message for UI. It will catch validation error and return `HTTP 400 Bad request` for API.

Reply:
Should we do it that way, or code the Http error codes directly into an Exception base class we generally agree to use? I understand the reflexive appeal of translating from a HTTP-unaware exception into a HTTP return code somewhere in the HTTP code, but as the years go on, I wonder if this was a real economy in any way, or whether it's reasonable to add optional HTTP information -- including exposed error message -- to concrete exceptions in the system raised pretty much anywhere.

This is not an all or nothing type of design choice, in some cases the class/base class says enough, but it's a somewhat unorthodox thought I've been having for a while now.